### PR TITLE
Replace environment variables with arguments

### PR DESCRIPTION
**Pull request by stshrive**
Add simple argument parser

### CODE REVIEW

**Comment by stshrive** [comment]
@paulgmiller, this requires a change to the release definition for it to run. This branch is currently hosted at [stshriv/vstsdropdownloadcore](https://hub.docker.com/r/stshriv/vstsdropdownloadcore/) on docker hub

File: Args.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        public string VstsPat { get; set; }
+
+        public string DropDestination { get; set; }
+
+        public string DropUrl { get; set; }
+
+        public string RelativePath { get; set; }
+
+        public void Parse(string[] args)
```
Comment (paulgmiller) [inline Args.cs]: any reason you didn't just use. https://www.nuget.org/packages/CommandLineParser/

File: Program.cs [hunk 2]
```diff
@@ -8,26 +8,23 @@ namespace DropDownloadCore
...
-        private const string DropDestinationEnvironmentVariable = "dropdestination";
         private const string DefaultDropDestination = "/drop";
-        private const string DropUrlEnvironmentVariable = "dropurl";
-
      
         static void Main(string[] args)
         {
-            var relativePath = System.Environment.GetEnvironmentVariable(RelavePathEnvironmentVariable) ?? "/";
-            var pat = System.Environment.GetEnvironmentVariable(VSTSPatEnvironmentVariable);
+            dropdownloadcore.Args arguments = new dropdownloadcore.Args();
```
Comment (paulgmiller) [inline Program.cs]: we can just default these to the env vars

**Review by paulgmiller — CHANGES_REQUESTED** [review]
fine with what you're doing except I don't like custom arg parsign and want to fall back to default.

**Comment by stshrive** [comment]
I din't wan't to pull in a dependency on a full argument parser. It can be removed if you'd like.

File: Args.cs [hunk 3]
```diff
@@ -12,80 +18,136 @@ public class Args
...
-            shorthandArg = shorthandArg.Trim(trimChars);
-
-            bool shortEq = false;
-            if (!string.IsNullOrWhiteSpace(shorthandArg))
-            {
-                shortEq = givenArgument.Equals(string.Format("-{0}", shorthandArg));
-            }
-
-            return shortEq || givenArgument.Equals(string.Format("--{0}", expected), StringComparison.OrdinalIgnoreCase);
+        public void CheckEnvironmentVariables()
```
Comment (paulgmiller) [inline Args.cs]: so you could not do the arg checks if you instead make this a static Args FromEnvVars() {} method and use it to create your args.

Comment (paulgmiller) [reply Args.cs]: then remove it from parse

File: Args.cs [hunk 4]
```diff
@@ -12,80 +18,136 @@ public class Args
...
+                        this.RelativePath = args[i + 1];
+                        ++i;
+                        continue;
                     }
                 }
+
             }
+
+            this.CheckEnvironmentVariables();
+            this.ValidatePat();
```
Comment (paulgmiller) [inline Args.cs]: maybe shouldn't be part of parse?

**Review by paulgmiller — APPROVED** [review]

**Pull request merged**